runner-manager/cfd/cloudgov/cloudgov.go

```diff
@@ -87,3 +98,57 @@ func (c *Client) AppDelete(id string) error {
 func (c *Client) AppsList() ([]*App, error) {
 	return c.appsList()
 }
+
+// TODO: this abstraction might belong in /cmd,
+// unless it can be further generalized to all pushes
+func (c *Client) ServicePush(manifest *AppManifest) (*App, error) {
+	containerID := manifest.Name
+
+	if containerID == "" {
+		return nil, CloudGovClientError{"ServicePush: AppManifest.Name must be defined"}
+	}
+
+	if manifest.OrgName == "" || manifest.SpaceName == "" {
+		return nil, CloudGovClientError{"ServicePush: AppManifest must have Org and Space names"}
+	}
+
+	// check for an old instance of the service, delete if found
+	app, err := c.AppGet(containerID)
+	if err != nil {
+		var cferr resource.CloudFoundryError
+		if errors.As(err, &cferr) {
+			err = nil
+			if cferr.Code != 10010 {
+				return nil, fmt.Errorf("unexpected cferr checking for existing app: %w", cferr)
+			}
+		} else {
+			return nil, fmt.Errorf("error checking for existing service (%v): %w", containerID, err)
+		}
+	}
+	if app != nil {
+		err = c.AppDelete(containerID)
+	}
+	if err != nil {
+		return nil, fmt.Errorf("error deleting existing service (%v): %w", containerID, err)
+	}
```

Comment:
[suggestion (non-blocking)] it felt weird to be checking `err` twice in three top-level if statements.

```suggestion
	if app != nil {
		if err := c.AppDelete(containerID); err != nil {
			return nil, fmt.Errorf("error deleting existing service (%v): %w", containerID, err)
		}
	}
```